### PR TITLE
feat: walletconnect connector

### PR DESCRIPTION
**Pull request by pedropereiradev**

### CODE REVIEW

File: examples/react-app/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+VITE_APP_WC_PROJECT_ID=0e0f5503e675e719c07e73ff5f38d31f
```
Comment (luizstacio) [inline examples/react-app/.env:1]: we need to ignore this file from github

File: examples/react-app/src/main.tsx [hunk 2]
```diff
@@ -13,15 +23,56 @@ import App from './App.tsx';
 import ScreenSizeIndicator from './components/screensize-indicator.tsx';
 import './index.css';
 
-const queryClient = new QueryClient();
+const _queryClientt = new QueryClient();
```
Comment (luizstacio) [inline examples/react-app/src/main.tsx:26]:
```suggestion
const queryClient = new QueryClient();
```

File: packages/evm-connector/src/EvmWalletConnector.ts [hunk 3]
```diff
@@ -68,13 +68,16 @@ export class EVMWalletConnector extends FuelConnector {
   async configProviders(config: EVMWalletConnectorConfig = {}) {
     this.config = Object.assign(config, {
       fuelProvider: config.fuelProvider || Provider.create(BETA_5_URL),
-      ethProvider: config.ethProvider || window.ethereum,
+      ethProvider:
+        //@ts-ignore
```
Comment (luizstacio) [inline packages/evm-connector/src/EvmWalletConnector.ts:72]: Why you need this `@ts-ignore`?

File: packages/walletconnect-connector/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  //session
+  SESSION_EVENT = 'session_event',
+
+  //connection uri
+  DISPLAY_URI = 'display_uri',
+
+  //connections
+  CONNECT = 'connect',
+  DISCONNECT = 'disconnect',
+}
```
Comment (luizstacio) [inline packages/walletconnect-connector/src/types.ts:32]: ??

**Review by luizstacio — CHANGES_REQUESTED** [review]
Also the tests are failing because of lack of coverage, I understand it can be difficult to mock the walletconnect modal, if you don't find a ready solution, you can reduce the trashhold of the test coverage for the specific package.

File: examples/react-app/src/main.tsx [hunk 5]
```diff
@@ -15,13 +25,54 @@ import './index.css';
...
+// ============================================================
+const WC_PROJECT_ID = import.meta.env.VITE_APP_WC_PROJECT_ID;
+const METADATA = {
+  name: 'Wallet Demo',
+  description: 'Fuel Wallets Demo',
+  url: location.href,
+  icons: ['https://connectors.fuel.network/logo_white.png'],
+};
+const wagmiConfig = createConfig({
+  chains: [mainnet],
```
Comment (luizstacio) [inline examples/react-app/src/main.tsx:40]: Can you add here sepolia also?

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Review by HoTandy — COMMENTED** [review]
I've raised the following issues here:
https://github.com/FuelLabs/fuel-connectors/issues/54
https://github.com/FuelLabs/fuel-connectors/issues/53

With the Ethereum Wallets connection option, I'm finding it virtually impossible to connect to Trust wallet or Coinbase wallet with the Sepolia network. I'm wondering if this is a problem on their side though rather than on the Fuel side.
I think I had this problem when I first started to test Connectors and then tried to Connect Metamask, ...it ended up that a fix was possible on the Fuel side. (https://github.com/FuelLabs/fuel-connectors/issues/35)
Will it be the same with these other wallets?  Linking up with mainnets is never an issue.

WalletConnect I can scan with the camera on my iPhone (I don't have an Android phone), it always wants to open Metamask which I would kind of expect with iOS.  It's always impossible to change the default app on iOS.
I can scan the WalletConnect QR codes from within different wallet apps though and each time is seems to work ok, except it always want to connect Mainnet addresses to the Fuel Connector.
I cannot imagine that this is something that can be changed on the Fuel side, or is it?

**Comment by luizstacio** [comment]
close in favor of #61

**Pull request closed without merging**